src/LanguageServer.Engine/CompletionProviders/CompletionProvider.cs

```diff
@@ -94,13 +94,18 @@ protected virtual bool HandleTriggerCharacters(string triggerCharacters, Project
                 throw new ArgumentNullException(nameof(projectDocument));
 
             // Replace any characters that were typed to trigger the completion.
-            if (triggerCharacters != null)
+            if (!String.IsNullOrEmpty(triggerCharacters))
             {
-                targetRange = projectDocument.XmlPositions.ExtendLeft(targetRange, byCharCount: triggerCharacters.Length);
+                // The last character typed is implicitly part of the current selection, if it triggered completion.
+                int extendSelectionByCharCount = triggerCharacters.Length - 1;
```

Comment:
I wonder whether it is even possible to get more than 1 trigger char. Yes, omnisharp defines this as `string`, but it is trigger _character_ after all. No changes required here, we will be safer if we compute this offset properly